index.js

```diff
@@ -218,21 +217,19 @@ function _buildUrl(config, type, multi) {
   return `${config.url}${block}?v=${version}&l=${language}`
 }
 
-function fetchUrl(urlIn, config, blockName) {
+async function fetchBlock(urlIn, config, blockName) {
   const headers = config.headers ? config.headers : {}
-  return fetch(urlIn, { headers })
-    .then(async result => {
-      if (result.ok) {
-        const text = await result.text()
-        const rval = { blockName, result: text }
-        return rval
-      }
+  try {
+    const response = await fetch(urlIn, { headers })
+    if (!response.ok) {
       log.error(`Failed to fetch cortina block at ${urlIn}: ${result.status}`)
       return { blockName, result: '' }
-    })
-    .catch(err => {
-      log.error(`WARNING! FAILED TO FETCH ${blockName} ${err.toString()}`)
-    })
+    }
+    const result = await response.text()
+    return { blockName, result }
+  } catch (err) {
+    log.error(`WARNING! FAILED TO FETCH ${blockName} ${err.toString()}`)
```

Comment:
Catch block should also return?

Reply:
Hmm It thought I could skip the return in catch since we are checking if returned block i defined on line 269, but maybe better to return it as it was done earlier?

Reply:
Sure. ESLint reports ”Expected to return a value at the end of async function 'fetchBlock'.” due to the inconsistent returns. Maybe a little clearer to return `null`, instead of just letting it fizzle out and return `undefined`?

Also, what’s the philosophical difference between a `!response.ok` and a thrown error? One returns an object, and the other doesn’t?

Reply:
Ah, I missed the eslint warnings 😅

Reply:
And yes I agree! We should be consistent with returns 🙂